README.md

```diff
@@ -412,57 +410,37 @@ The AdView is responsible for:
 1. Pulling all possible demand (campaigns, bids) from the [Market](#market);
 2. Picking which ad to show depending on the user: this depends on a combination of price and targeting (header bidding and contextual targeting);
 3. Generating events (impressions as per [IAB's guidelines](https://www.iab.com/wp-content/uploads/2015/06/Ad-Impression-Measurment-Guideline-US.pdf), clicks) and sending them to all validators and observers of the given ad;
+4. Comparing the generated data with that provided by the SSP and pinpointing any discrepancies
 
 Later on, if needed, it will also be responsible for:
 
-1. Creating a cryptographic identity (keypair) for the user, if they don't already have one, and persisting it in their browser;
+Creating a cryptographic identity (keypair) for the user, if they don't already have one, and persisting it in their browser;
+
 
 #### Contextual targeting
 
-Notice a common pattern here: **sensitive information never leaves the user's browser**, and this is achieved by shifting the process of targeting (selecting ads) to the browser itself. To achieve this, we use [contextual targeting](https://medium.com/the-adex-blog/why-we-use-contextual-targeting-d49f3ecf0acf).
+Notice a common pattern here: **sensitive information never leaves the user's browser**, and this is achieved by shifting the process of targeting (selecting ads) to the browser itself. To achieve this, we use [contextual targeting](https://medium.com/the-adex-blog/why-we-use-contextual-targeting-d49f3ecf0acf)for our direct publishers. When it comes to SSPs, the targeting and data collection is done by them.
```

Comment:
there's no space after "contextual advertising". I would add "via OpenRTB" at the end